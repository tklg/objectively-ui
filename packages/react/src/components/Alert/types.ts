import { ForwardRefExoticComponent, ReactElement } from 'react'
import { PropsWithChildrenAndClassName } from 'src/types/PropsWithChildrenAndClassName'
import { StatusColor } from 'src/types/props'

export interface AlertProps extends PropsWithChildrenAndClassName {
  severity?: StatusColor;
  title?: string;
  action?: ReactElement;
}
export type AlertComponent<P = AlertProps> = ForwardRefExoticComponent<P>
