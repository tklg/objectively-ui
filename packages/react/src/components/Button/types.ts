import { ButtonHTMLAttributes, DetailedHTMLProps, ReactNode } from 'react'
import { CommonSize } from 'src/types/sizes'

export type ButtonVariant = 'default' | 'primary' | 'text' | 'link';

export interface ButtonProps extends DetailedHTMLProps<ButtonHTMLAttributes<HTMLButtonElement>, HTMLButtonElement> {
  size?: CommonSize;
  fullWidth?: boolean;
  iconStart?: ReactNode;
  variant?: ButtonVariant;
  danger?: boolean;
}
