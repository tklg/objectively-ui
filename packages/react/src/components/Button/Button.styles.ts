import { css } from '@emotion/react'
import { focusOutlineStylesWithoutElement } from 'src/components/Accessibility.styles'
import { ColorTheme } from 'src/types/ColorTheme'
import { PROJECT_SHORTNAME } from 'src/utils/constants'

export const buttonStyles = (theme: ColorTheme) => css({
  position: 'relative',
  display: 'inline-block',
  borderWidth: 0,
  outlineOffset: 0,
  borderStyle: 'solid',
  borderColor: theme.colors.border,
  outlineColor: theme.colors.border,
  padding: `0 ${theme.spacing.md}`,
  minHeight: theme.size.md,
  background: 'transparent',
  color: theme.colors.textPrimary,
  borderRadius: theme.radii.md,
  cursor: 'pointer',
  userSelect: 'none',
  transition: [
    `background ${theme.transitions.duration.fast} ${theme.transitions.function.default}`,
    `transform ${theme.transitions.duration.fast} ${theme.transitions.function.default}`,
  ].join(','),

  '&:hover': {
    background: theme.colors.hover,
  },

  '&:active': {
    background: theme.colors.active,
    transform: 'scale(0.97)',
  },

  '&:focus': {
    outline: 'none',
  },

  '&:focus-visible:before': {
    content: '\'\'',
    position: 'absolute',
    height: '100%',
    width: '100%',
    top: 0,
    left: 0,
    ...focusOutlineStylesWithoutElement(theme),
  },

  [`&.${PROJECT_SHORTNAME}-Button-sizeSM`]: {
    minHeight: theme.size.sm,
  },
  [`&.${PROJECT_SHORTNAME}-Button-sizeLG`]: {
    minHeight: theme.size.lg,
  },
  [`&.${PROJECT_SHORTNAME}-Button-fullWidth`]: {
    display: 'block',
    width: '100%',
  },

  [`&.${PROJECT_SHORTNAME}-Button-variantLink`]: {
    borderColor: 'transparent',
    color: theme.colors.link,
  },

  [`&.${PROJECT_SHORTNAME}-Button-variantDefault`]: {
    borderWidth: theme.lines.sm,
    outlineOffset: theme.lines.sm,

    ...['Primary', 'Secondary', 'Info', 'Warning', 'Error', 'Success'].reduce((a, status) => {
      const key = status === 'Primary' || status === 'Secondary'
        ? (`accent${status}` as 'accentPrimary' | 'accentSecondary')
        : (`status${status}` as 'statusInfo' | 'statusWarning' | 'statusError' | 'statusSuccess')
      return {
        ...a,
        [`&.${PROJECT_SHORTNAME}-Button-color${status}`]: {
          color: theme.colors[key].value,
          borderColor: theme.colors[key].value,
          outlineColor: theme.colors[key].value,

          '&:hover': {
            background: theme.colors.hover,
            borderColor: theme.colors[key].hover,
          },

          '&:active': {
            background: theme.colors.active,
            borderColor: theme.colors[key].active,
          },
        },
      }
    }, {}),
  },
  [`&.${PROJECT_SHORTNAME}-Button-variantText`]: {
    borderColor: 'transparent',

    ...['Primary', 'Secondary', 'Info', 'Warning', 'Error', 'Success'].reduce((a, status) => {
      const key = status === 'Primary' || status === 'Secondary'
        ? (`accent${status}` as 'accentPrimary' | 'accentSecondary')
        : (`status${status}` as 'statusInfo' | 'statusWarning' | 'statusError' | 'statusSuccess')
      return {
        ...a,
        [`&.${PROJECT_SHORTNAME}-Button-color${status}`]: {
          color: theme.colors[key].value,
          outlineColor: theme.colors[key].value,

          '&:hover': {
            background: theme.colors.hover,
          },

          '&:active': {
            background: theme.colors.active,
          },
        },
      }
    }, {}),
  },
  [`&.${PROJECT_SHORTNAME}-Button-variantSolid`]: {
    background: theme.colors.border,
    color: theme.colors.textPrimary,
    borderColor: theme.colors.border,
    outlineColor: theme.colors.border,

    '&:hover': {
      background: theme.colors.hover,
      borderColor: theme.colors.hover,
    },

    '&:active': {
      background: theme.colors.active,
      borderColor: theme.colors.active,
    },

    ...['Primary', 'Secondary', 'Info', 'Warning', 'Error', 'Success'].reduce((a, status) => {
      const key = status === 'Primary' || status === 'Secondary'
        ? (`accent${status}` as 'accentPrimary' | 'accentSecondary')
        : (`status${status}` as 'statusInfo' | 'statusWarning' | 'statusError' | 'statusSuccess')
      return {
        ...a,
        [`&.${PROJECT_SHORTNAME}-Button-color${status}`]: {
          background: theme.colors[key].value,
          color: theme.colors[key].contrastText,
          borderColor: theme.colors[key].value,
          outlineColor: theme.colors[key].value,

          '&:hover': {
            background: theme.colors[key].hover,
            borderColor: theme.colors[key].hover,
          },

          '&:active': {
            background: theme.colors[key].active,
            borderColor: theme.colors[key].active,
          },
        },
      }
    }, {}),
  },
})

export const buttonGlowStyles = (theme: ColorTheme) => css({
  position: 'absolute',
  height: '100%',
  width: '100%',
  top: 0,
  left: 0,
  outlineOffset: 'inherit',
  outline: `${theme.lines.lg} solid`,
  outlineColor: 'inherit',
  borderRadius: 'inherit',

  animation: `${theme.transitions.duration.slow} ${theme.transitions.function.default} button-outline-fadeout`,
  animationFillMode: 'forwards',

  '@keyframes button-outline-fadeout': {
    from: {
      opacity: 1,
      outlineWidth: 0,
    },
    to: {
      opacity: 0,
      outlineWidth: `calc(${theme.lines.lg} * 1.75)`,
    },
  },
})
