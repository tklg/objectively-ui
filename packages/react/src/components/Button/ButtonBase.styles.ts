import { css } from '@emotion/react'
import { focusOutlineStylesWithoutElement } from 'src/components/Accessibility.styles'
import { ColorTheme } from 'src/types/ColorTheme'
import { PROJECT_SHORTNAME } from 'src/utils/constants'

export const buttonBaseStyles = (theme: ColorTheme) => css({
  position: 'relative',
  display: 'inline-flex',
  alignItems: 'center',
  justifyContent: 'center',
  borderWidth: 0,
  outlineOffset: 0,
  borderStyle: 'solid',
  borderColor: theme.colors.border,
  outlineColor: theme.colors.border,
  padding: `0 ${theme.spacing.md}`,
  minHeight: theme.size.md,
  borderRadius: theme.radii.md,
  background: 'transparent',
  color: theme.colors.textPrimary,
  cursor: 'pointer',
  userSelect: 'none',
  verticalAlign: 'middle',
  transition: [
    `background ${theme.transitions.duration.fast} ${theme.transitions.function.default}`,
    `transform ${theme.transitions.duration.fast} ${theme.transitions.function.default}`,
  ].join(','),

  '&:hover:not(:disabled)': {
    background: theme.colors.hover,
  },

  '&:active:not(:disabled)': {
    background: theme.colors.active,
    transform: 'scale(0.97)',
  },

  '&:disabled': {
    cursor: 'default',
  },

  '&:focus': {
    outline: 'none',
  },

  '&:focus-visible:before': {
    content: '\'\'',
    position: 'absolute',
    height: '100%',
    width: '100%',
    top: 0,
    left: 0,
    ...focusOutlineStylesWithoutElement(theme),
  },

  [`&.${PROJECT_SHORTNAME}-ButtonBase-sizeSm`]: {
    minHeight: theme.size.sm,
    borderRadius: theme.radii.sm,
  },
  [`&.${PROJECT_SHORTNAME}-ButtonBase-sizeLg`]: {
    minHeight: theme.size.lg,
    borderRadius: theme.radii.lg,
  },

  [`&.${PROJECT_SHORTNAME}-ButtonBase-variantDefault`]: {
    borderWidth: theme.lines.sm,
    outlineOffset: theme.lines.sm,
    '&:disabled': {
      color: `${theme.colors.textDisabled} !important`,
    },

    ...['Primary', 'Secondary', 'Info', 'Warning', 'Error', 'Success'].reduce((a, status) => {
      const key = status === 'Primary' || status === 'Secondary'
        ? (`accent${status}` as 'accentPrimary' | 'accentSecondary')
        : (`status${status}` as 'statusInfo' | 'statusWarning' | 'statusError' | 'statusSuccess')
      return {
        ...a,

        [`&.${PROJECT_SHORTNAME}-ButtonBase-color${status}`]: {
          color: theme.colors[key].value,
          borderColor: theme.colors[key].value,
          outlineColor: theme.colors[key].value,

          '&:hover:not(:disabled)': {
            background: theme.colors.hover,
            borderColor: theme.colors[key].hover,
          },

          '&:active:not(:disabled)': {
            background: theme.colors.active,
            borderColor: theme.colors[key].active,
          },

          '&:disabled': {
            background: theme.colors.backgroundDisabled,
            borderColor: theme.colors[key].shadow,
          },
        },
      }
    }, {}),
  },
  [`&.${PROJECT_SHORTNAME}-ButtonBase-variantText`]: {
    borderColor: 'transparent',

    '&:disabled': {
      color: theme.colors.textDisabled,
    },

    ...['Primary', 'Secondary', 'Info', 'Warning', 'Error', 'Success'].reduce((a, status) => {
      const key = status === 'Primary' || status === 'Secondary'
        ? (`accent${status}` as 'accentPrimary' | 'accentSecondary')
        : (`status${status}` as 'statusInfo' | 'statusWarning' | 'statusError' | 'statusSuccess')
      return {
        ...a,

        [`&.${PROJECT_SHORTNAME}-ButtonBase-color${status}`]: {
          color: theme.colors[key].value,
          outlineColor: theme.colors[key].value,

          '&:hover:not(:disabled)': {
            background: theme.colors.hover,
          },

          '&:active:not(:disabled)': {
            background: theme.colors.active,
          },

          '&:disabled': {
            borderColor: theme.colors[key].shadow,
            color: theme.colors[key].shadow,
          },
        },
      }
    }, {}),
  },
  [`&.${PROJECT_SHORTNAME}-ButtonBase-variantSolid`]: {
    background: theme.colors.border,
    color: theme.colors.textPrimary,
    borderColor: theme.colors.border,
    outlineColor: theme.colors.border,

    '&:hover:not(:disabled)': {
      background: theme.colors.hover,
      borderColor: theme.colors.hover,
    },

    '&:active:not(:disabled)': {
      background: theme.colors.active,
      borderColor: theme.colors.active,
    },

    '&:disabled': {
      color: theme.colors.textDisabled,
    },

    ...['Primary', 'Secondary', 'Info', 'Warning', 'Error', 'Success'].reduce((a, status) => {
      const key = status === 'Primary' || status === 'Secondary'
        ? (`accent${status}` as 'accentPrimary' | 'accentSecondary')
        : (`status${status}` as 'statusInfo' | 'statusWarning' | 'statusError' | 'statusSuccess')
      return {
        ...a,

        [`&.${PROJECT_SHORTNAME}-ButtonBase-color${status}`]: {
          background: theme.colors[key].value,
          color: theme.colors[key].contrastText,
          borderColor: theme.colors[key].value,
          outlineColor: theme.colors[key].value,

          '&:hover:not(:disabled)': {
            background: theme.colors[key].hover,
            borderColor: theme.colors[key].hover,
          },

          '&:active:not(:disabled)': {
            background: theme.colors[key].active,
            borderColor: theme.colors[key].active,
          },

          '&:disabled': {
            background: theme.colors[key].shadow,
            borderColor: theme.colors[key].shadow,
            color: theme.colors[key].contrastTextDisabled,
          },
        },
      }
    }, {}),
  },
})

export const buttonBaseGlowStyles = (theme: ColorTheme) => css({
  position: 'absolute',
  height: '100%',
  width: '100%',
  top: 0,
  left: 0,
  outlineOffset: 'inherit',
  outline: `${theme.lines.lg} solid`,
  outlineColor: 'inherit',
  borderRadius: 'inherit',
  pointerEvents: 'none',

  animation: `${theme.transitions.duration.slow} ${theme.transitions.function.default} button-outline-fadeout`,
  animationFillMode: 'forwards',

  '@keyframes button-outline-fadeout': {
    from: {
      opacity: 1,
      outlineWidth: 0,
    },
    to: {
      opacity: 0,
      outlineWidth: `calc(${theme.lines.lg} * 1.75)`,
    },
  },
})
