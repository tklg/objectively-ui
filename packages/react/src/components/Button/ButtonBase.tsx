import { forwardRef, useCallback, useState } from 'react'
import { buttonBaseGlowStyles, buttonBaseStyles } from 'src/components/Button/ButtonBase.styles'
import { ButtonBaseComponent, ButtonBaseProps } from 'src/components/Button/ButtonBase.types'
import { useTheme } from 'src/hooks'
import { buildClassName } from 'src/utils/buildClassName'

const ELEMENT_NAME = 'ButtonBase'

export const ButtonBase = forwardRef<HTMLButtonElement, ButtonBaseProps>(({
  children,
  size = 'md',
  color = 'default',
  variant,
  onMouseDown,
  onMouseUp,
  disabled,
  className: _className,
  ...props
}, ref) => {
  const theme = useTheme()
  const [mouseUpGlow, setMouseUpGlow] = useState(false)

  const className = buildClassName(ELEMENT_NAME, {
    size,
    color,
    variant,
  }, _className)

  const handleMouseDown = useCallback((e: React.MouseEvent<HTMLButtonElement, MouseEvent>) => {
    onMouseDown?.(e)
    setMouseUpGlow(false)
  }, [onMouseDown])

  const handleMouseUp = useCallback((e: React.MouseEvent<HTMLButtonElement, MouseEvent>) => {
    onMouseUp?.(e)
    setMouseUpGlow(true)
  }, [onMouseUp])

  return (
    <button
      ref={ref}
      {...props}
      disabled={disabled}
      className={className}
      css={buttonBaseStyles(theme)}
      onMouseDown={handleMouseDown}
      onMouseUp={handleMouseUp}
    >
      {!disabled && mouseUpGlow && <div css={buttonBaseGlowStyles(theme)} />}
      {children}
    </button>
  )
}) as ButtonBaseComponent

ButtonBase.displayName = ELEMENT_NAME

if (process.env.NODE_ENV !== 'production') {
  ButtonBase.displayName = ELEMENT_NAME
}
