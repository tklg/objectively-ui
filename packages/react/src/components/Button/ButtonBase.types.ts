import { ButtonHTMLAttributes, DetailedHTMLProps, ForwardRefExoticComponent } from 'react'
import { CommonColor, CommonSize } from 'src/types/props'

export interface ButtonBaseProps extends DetailedHTMLProps<ButtonHTMLAttributes<HTMLButtonElement>, HTMLButtonElement> {
  size?: CommonSize;
  color?: CommonColor;
  variant?: string;
}

export type ButtonBaseComponent<P = ButtonBaseProps> = ForwardRefExoticComponent<P>
