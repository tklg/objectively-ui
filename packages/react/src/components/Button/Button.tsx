import { forwardRef, useCallback, useEffect, useState } from 'react'
import { buttonGlowStyles, buttonStyles } from 'src/components/Button/Button.styles'
import { ButtonProps } from 'src/components/Button/types'
import { useTheme } from 'src/hooks'
import { buildClassName } from 'src/utils/buildClassName'

const ELEMENT_NAME = 'Button'

export const Button = forwardRef<HTMLButtonElement, ButtonProps>(({
  children,
  size = 'md',
  variant = 'default',
  danger,
  fullWidth,
  onMouseDown,
  onMouseUp,
  className: _className,
  ...props
}, ref) => {
  const theme = useTheme()
  const [mouseUpGlow, setMouseUpGlow] = useState(false)

  const className = buildClassName(ELEMENT_NAME, {
    size: size.toUpperCase(),
    variant,
    fullWidth,
    danger,
  }, _className)

  const handleMouseDown = useCallback((e: React.MouseEvent<HTMLButtonElement, MouseEvent>) => {
    onMouseDown?.(e)
    setMouseUpGlow(false)
  }, [onMouseDown])

  const handleMouseUp = useCallback((e: React.MouseEvent<HTMLButtonElement, MouseEvent>) => {
    onMouseUp?.(e)
    setMouseUpGlow(true)
  }, [onMouseUp])

  useEffect(() => {
    let timeout: number
    if (mouseUpGlow) {
      // timeout = window.setTimeout(() => setMouseUpGlow(false), 1000)
    }
    return () => window.clearTimeout(timeout)
  }, [mouseUpGlow])

  return (
    <button
      ref={ref}
      {...props}
      className={className}
      css={buttonStyles(theme)}
      onMouseDown={handleMouseDown}
      onMouseUp={handleMouseUp}
    >
      {mouseUpGlow && <div css={buttonGlowStyles(theme)} />}
      <span>{children}</span>
    </button>
  )
})

Button.displayName = ELEMENT_NAME
