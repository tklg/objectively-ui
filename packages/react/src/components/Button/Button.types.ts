import { ForwardRefExoticComponent, ReactNode } from 'react'
import { ButtonBaseProps } from 'src/components/Button/ButtonBase.types'

export type ButtonVariant = 'default' | 'solid' | 'text' | 'link'

export interface ButtonProps extends ButtonBaseProps {
  fullWidth?: boolean;
  iconStart?: ReactNode;
  iconEnd?: ReactNode;
  variant?: ButtonVariant;
}

export type ButtonComponent<P = ButtonProps> = ForwardRefExoticComponent<P>
