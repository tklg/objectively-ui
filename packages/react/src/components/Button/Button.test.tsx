import { Button } from 'src/components/Button/Button'
import { ButtonVariant } from 'src/components/Button/Button.types'
import { CommonColor, CommonSize } from 'src/types/props'
import { render, screen } from 'src/__test__/render'

describe('Button', () => {
  it('renders a button', () => {
    render(<Button>text</Button>)
    expect(screen.getByRole('button')).toHaveTextContent('text')
  })

  it('renders a disabled button', () => {
    render(<Button disabled>text</Button>)
    expect(screen.getByRole('button')).toBeDisabled()
  })

  it('with a start icon', () => {
    render(<Button iconStart='icon'>text</Button>)
    expect(screen.getByRole('button')).toHaveTextContent('icon')
    expect(screen.getByRole('button')).toMatchSnapshot()
  })

  it('with an end icon', () => {
    render(<Button iconEnd='icon'>text</Button>)
    expect(screen.getByRole('button')).toHaveTextContent('icon')
    expect(screen.getByRole('button')).toMatchSnapshot()
  })

  describe('sizes', () => {
    it.each([['sm'], ['md'], ['lg']])('%s', (args) => {
      const size = args[0] as CommonSize

      render(<Button size={size}>text</Button>)
      expect(screen.getByRole('button')).toMatchSnapshot(size)
    })
  })

  describe('variants', () => {
    it.each([['default'], ['solid'], ['text'], ['link']])('%s', (args) => {
      const variant = args[0] as ButtonVariant

      render(<Button variant={variant}>text</Button>)
      expect(screen.getByRole('button')).toMatchSnapshot(variant)
    })
  })

  describe('colors', () => {
    it.each([['default'], ['primary'], ['secondary'], ['info'], ['warning'], ['error'], ['success']])('%s', (args) => {
      const color = args[0] as CommonColor

      render(<Button color={color}>text</Button>)
      expect(screen.getByRole('button')).toMatchSnapshot(color)
    })
  })
})
