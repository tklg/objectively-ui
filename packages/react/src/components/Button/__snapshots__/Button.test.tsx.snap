// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Button colors default: d 1`] = `
<button
  class="ObjUI-ButtonBase ObjUI-ButtonBase-sizeMd ObjUI-ButtonBase-colorD ObjUI-ButtonBase-variantDefault ObjUI-Button ObjUI-Button-sizeMd ObjUI-Button-variantDefault ObjUI-Button-colorD emotion-0"
>
  <span>
    text
  </span>
</button>
`;

exports[`Button colors error: e 1`] = `
<button
  class="ObjUI-ButtonBase ObjUI-ButtonBase-sizeMd ObjUI-ButtonBase-colorE ObjUI-ButtonBase-variantDefault ObjUI-Button ObjUI-Button-sizeMd ObjUI-Button-variantDefault ObjUI-Button-colorE emotion-0"
>
  <span>
    text
  </span>
</button>
`;

exports[`Button colors info: i 1`] = `
<button
  class="ObjUI-ButtonBase ObjUI-ButtonBase-sizeMd ObjUI-ButtonBase-colorI ObjUI-ButtonBase-variantDefault ObjUI-Button ObjUI-Button-sizeMd ObjUI-Button-variantDefault ObjUI-Button-colorI emotion-0"
>
  <span>
    text
  </span>
</button>
`;

exports[`Button colors primary: p 1`] = `
<button
  class="ObjUI-ButtonBase ObjUI-ButtonBase-sizeMd ObjUI-ButtonBase-colorP ObjUI-ButtonBase-variantDefault ObjUI-Button ObjUI-Button-sizeMd ObjUI-Button-variantDefault ObjUI-Button-colorP emotion-0"
>
  <span>
    text
  </span>
</button>
`;

exports[`Button colors secondary: s 1`] = `
<button
  class="ObjUI-ButtonBase ObjUI-ButtonBase-sizeMd ObjUI-ButtonBase-colorS ObjUI-ButtonBase-variantDefault ObjUI-Button ObjUI-Button-sizeMd ObjUI-Button-variantDefault ObjUI-Button-colorS emotion-0"
>
  <span>
    text
  </span>
</button>
`;

exports[`Button colors success: s 1`] = `
<button
  class="ObjUI-ButtonBase ObjUI-ButtonBase-sizeMd ObjUI-ButtonBase-colorS ObjUI-ButtonBase-variantDefault ObjUI-Button ObjUI-Button-sizeMd ObjUI-Button-variantDefault ObjUI-Button-colorS emotion-0"
>
  <span>
    text
  </span>
</button>
`;

exports[`Button colors warning: w 1`] = `
<button
  class="ObjUI-ButtonBase ObjUI-ButtonBase-sizeMd ObjUI-ButtonBase-colorW ObjUI-ButtonBase-variantDefault ObjUI-Button ObjUI-Button-sizeMd ObjUI-Button-variantDefault ObjUI-Button-colorW emotion-0"
>
  <span>
    text
  </span>
</button>
`;

exports[`Button sizes lg: l 1`] = `
<button
  class="ObjUI-ButtonBase ObjUI-ButtonBase-sizeL ObjUI-ButtonBase-colorDefault ObjUI-ButtonBase-variantDefault ObjUI-Button ObjUI-Button-sizeL ObjUI-Button-variantDefault ObjUI-Button-colorDefault emotion-0"
>
  <span>
    text
  </span>
</button>
`;

exports[`Button sizes md: m 1`] = `
<button
  class="ObjUI-ButtonBase ObjUI-ButtonBase-sizeM ObjUI-ButtonBase-colorDefault ObjUI-ButtonBase-variantDefault ObjUI-Button ObjUI-Button-sizeM ObjUI-Button-variantDefault ObjUI-Button-colorDefault emotion-0"
>
  <span>
    text
  </span>
</button>
`;

exports[`Button sizes sm: s 1`] = `
<button
  class="ObjUI-ButtonBase ObjUI-ButtonBase-sizeS ObjUI-ButtonBase-colorDefault ObjUI-ButtonBase-variantDefault ObjUI-Button ObjUI-Button-sizeS ObjUI-Button-variantDefault ObjUI-Button-colorDefault emotion-0"
>
  <span>
    text
  </span>
</button>
`;

exports[`Button variants default: d 1`] = `
<button
  class="ObjUI-ButtonBase ObjUI-ButtonBase-sizeMd ObjUI-ButtonBase-colorDefault ObjUI-ButtonBase-variantD ObjUI-Button ObjUI-Button-sizeMd ObjUI-Button-variantD ObjUI-Button-colorDefault emotion-0"
>
  <span>
    text
  </span>
</button>
`;

exports[`Button variants link: l 1`] = `
<button
  class="ObjUI-ButtonBase ObjUI-ButtonBase-sizeMd ObjUI-ButtonBase-colorDefault ObjUI-ButtonBase-variantL ObjUI-Button ObjUI-Button-sizeMd ObjUI-Button-variantL ObjUI-Button-colorDefault emotion-0"
>
  <span>
    text
  </span>
</button>
`;

exports[`Button variants solid: s 1`] = `
<button
  class="ObjUI-ButtonBase ObjUI-ButtonBase-sizeMd ObjUI-ButtonBase-colorDefault ObjUI-ButtonBase-variantS ObjUI-Button ObjUI-Button-sizeMd ObjUI-Button-variantS ObjUI-Button-colorDefault emotion-0"
>
  <span>
    text
  </span>
</button>
`;

exports[`Button variants text: t 1`] = `
<button
  class="ObjUI-ButtonBase ObjUI-ButtonBase-sizeMd ObjUI-ButtonBase-colorDefault ObjUI-ButtonBase-variantT ObjUI-Button ObjUI-Button-sizeMd ObjUI-Button-variantT ObjUI-Button-colorDefault emotion-0"
>
  <span>
    text
  </span>
</button>
`;

exports[`Button with a start icon 1`] = `
<button
  class="ObjUI-ButtonBase ObjUI-ButtonBase-sizeMd ObjUI-ButtonBase-colorDefault ObjUI-ButtonBase-variantDefault ObjUI-Button ObjUI-Button-sizeMd ObjUI-Button-variantDefault ObjUI-Button-colorDefault ObjUI-Button-iconStart emotion-0"
>
  icon
  <span>
    text
  </span>
</button>
`;

exports[`Button with an end icon 1`] = `
<button
  class="ObjUI-ButtonBase ObjUI-ButtonBase-sizeMd ObjUI-ButtonBase-colorDefault ObjUI-ButtonBase-variantDefault ObjUI-Button ObjUI-Button-sizeMd ObjUI-Button-variantDefault ObjUI-Button-colorDefault ObjUI-Button-iconEnd emotion-0"
>
  <span>
    text
  </span>
  icon
</button>
`;
